app/src/ui/changes/filter-changes-list.tsx

```diff
@@ -972,6 +1037,31 @@ export class FilterChangesList extends React.Component<
     this.includeAllCheckBoxRef.current?.focus()
   }
 
+  private onChangedFileClick = (
+    item: IChangesListItem,
+    source: ClickSource
+  ) => {
+    const fileIndex = this.props.workingDirectory.files.findIndex(
```

Comment:
Could we use `workingDirectory.findFileIndexByID` here? That's backed by a Map which gets us constant time lookup as opposed to linear time with findIndex.